batch/batch/driver/instance_collection/pool.py

```diff
@@ -428,7 +437,12 @@ async def create_instances(self):
             AUTOSCALER_HEAD_JOB_QUEUE_READY_CORES.labels(pool_name=self.name, region=region).set(ready_cores_mcpu)
 
     async def control_loop(self):
-        await periodically_call(AUTOSCALER_LOOP_PERIOD_SECONDS, self.create_instances)
```

Comment:
Without a clarifying comment, I could easily see myself changing this back to using `periodically_call` because I don't pick up on the subtlety that `self.autoscaler_loop_period_secs` is subject to change. I would either add an explicit comment here, write something like `periodically_call_dynamic_sleep(lambda: self.autoscaler_loop_period_secs, self.create_instances)`, or make `periodically_call` accept a `Union[int, Callable[[], int]]` and continue using `periodically_call`.